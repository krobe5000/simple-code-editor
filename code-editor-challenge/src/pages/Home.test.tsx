import React from "react";
import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import HomePage from "./Home";
import { MemoryRouter as Router } from "react-router";

const renderComponent = () => {
  return render(
    <Router>
      <HomePage />
    </Router>
  );
};

test("Should render the Home Page", () => {
  const container = renderComponent();

  expect(container).toMatchSnapshot();
});
