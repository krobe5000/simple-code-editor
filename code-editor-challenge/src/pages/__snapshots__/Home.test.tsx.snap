// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the Home Page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <nav
            aria-label="breadcrumb"
            class="MuiTypography-root MuiBreadcrumbs-root MuiTypography-body1 MuiTypography-colorTextSecondary"
          >
            <ol
              class="MuiBreadcrumbs-ol"
            >
              <li
                class="MuiBreadcrumbs-li"
              >
                <a
                  href="/"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
                  >
                    Home
                  </p>
                </a>
              </li>
              <li
                aria-hidden="true"
                class="MuiBreadcrumbs-separator"
              >
                /
              </li>
              <li
                class="MuiBreadcrumbs-li"
              >
                <a
                  href="/viewEditor"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
                  >
                    View Editor
                  </p>
                </a>
              </li>
            </ol>
          </nav>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3"
          >
            Home page
          </h3>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle1"
          >
            Welcome to the Code Editor!
          </h6>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <nav
          aria-label="breadcrumb"
          class="MuiTypography-root MuiBreadcrumbs-root MuiTypography-body1 MuiTypography-colorTextSecondary"
        >
          <ol
            class="MuiBreadcrumbs-ol"
          >
            <li
              class="MuiBreadcrumbs-li"
            >
              <a
                href="/"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
                >
                  Home
                </p>
              </a>
            </li>
            <li
              aria-hidden="true"
              class="MuiBreadcrumbs-separator"
            >
              /
            </li>
            <li
              class="MuiBreadcrumbs-li"
            >
              <a
                href="/viewEditor"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
                >
                  View Editor
                </p>
              </a>
            </li>
          </ol>
        </nav>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3"
        >
          Home page
        </h3>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle1"
        >
          Welcome to the Code Editor!
        </h6>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
